pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -623,6 +623,11 @@ func (nc *Controller) doNoExecuteTaintingPass(ctx context.Context) {
 				return false, 50 * time.Millisecond
 			}
 			_, condition := controllerutil.GetNodeCondition(&node.Status, v1.NodeReady)
+			if condition == nil {
```

Comment:
can you please add an unit test to cover this regression?

Reply:
https://github.com/kubernetes/kubernetes/pull/122874#issuecomment-1900512132
+1 on this.

Reply:
@kannon92 @aojea   I made an attempt. Please review it.